doc/source/examples/incompressible-flow/3d_nitsche_mixer_with_pbt_impeller/nitscher_mixer_with_pbt_impeller.rst

```diff
@@ -0,0 +1,279 @@
+
+
+We specify the ``solid velocity`` using a used defined function. In the present case, the velocity that we specify induces a rotation around the :math:`x` axis of the meshs, centered at the point :math:`(0,0,0)`. This corresponds to the location of the center of our PBT in the accompanying GMSH mesh.
+
+The ``beta`` parameter controls the intensity of the application of Nitsche method in the fluid region. Setting a higher value of ``beta`` leads to a stiffer problem, but prevents the fluid from penetrating the solid. For flow at Reynolds number larger than one, we have found that setting ``beta`` between 10 and 1000 leads to satisfactory results. When the flow is more viscous, higher values of ``beta`` should be used to compensate for the larger shear stresses acting on the immersed solid.
+
+Nitsche method can be used with both static and dynamic immersed solid. When the velocity imposed on the solid leads to a motion of the solid, we must set ``enable particles motion=true`` . This option should only be set to true on the solids which are actually moving, since it adds a non-negligible cost to the simulations. To ensure that particles are always located efficiently as they move through cell, we set ``particles sub iterations = 10``. This enables us to displace the particle using 10 sub-steps, thus facilitating the detection of the cells in which they lie. Generally, it is a good practice to have sufficient particles sub iterations so as to ensure that particles do not move more than half a cell per particle sub iteration.
+
+
+Post-processing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When simulating a transient flow, it is often desirable to obtain the time-averaged velocity field. This is achieved through the post-processing section. The averaging period is started after two seconds, a time after which the flow has started to stabilize and reach pseudo steady-state. This is choice is error prone, but increasing the duration of the simulation and, consequently, of the averaging period, can compensate for an ill-chosen initial time for the averaging procedure.
+
+.. code-block:: text
+
+  subsection post-processing
+     set calculate average velocities = true
+     set initial time = 2
+  end
+
+Non-linear solver parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the problem that we solve is transient, it is not highly non-linear except for the first few iterations. Thus, we use pretty standard non-linear solver parameters.
+
+
+.. code-block:: text
+
+  subsection non-linear solver
+    set tolerance               = 1e-4
+    set max iterations          = 10
+    set verbosity               = verbose
+    set solver                  = inexact_newton
+  end
+
+Because the assembly of the Nitsche restriction for the immersed boundary is relatively expensive, we try to minimize it's assembly by reusing the Jacobian matrix as much as possible. This is accomplished setting ``solver=inexact_newton``.
```

Comment:
```suggestion
Because the assembly of the Nitsche restriction for the immersed boundary is relatively expensive, we try to minimize its assembly by reusing the Jacobian matrix as much as possible. This is accomplished setting ``solver=inexact_newton``.
```